### ISSUE TITLE
Splits multi char variables in \mathrm and \mathit into one <mi> for each letter

### ISSUE THREAD

**Issue by Leon-Lj** [ronkok/Temml#59]
If you have a multi character variable wrapped in `\mathit` or `\mathrm` the resulting mathml will not group it in one <mi> as expected, but rather split it into one <mi> for each letter. This causes problems for screen readers, which will read them one letter at a time, rather than the whole word.

For example `3\mathit{apples}+2\mathit{pears}=20\mathrm{kr}` should produce:

```
<math display="block">
  <mn>3</mn>
  <mrow">
    <mi>apples</mi>
  </mrow>
  <mo>+</mo>
  <mn>2</mn>
  <mrow>
    <mi>pears</mi>
  </mrow>
  <mo>=</mo>
  <mn>20</mn>
  <mrow>
    <mi>kr</mi>
  </mrow>
</math>
```

But Temml produces:
```
<math display="block" class="tml-display" style="display:block math;">
  <mrow>
    <mn>3</mn>
    <mrow>
      <mi>a</mi>
      <mi>p</mi>
      <mi>p</mi>
      <mi>l</mi>
      <mi>e</mi>
      <mi>s</mi>
    </mrow>
    <mo>+</mo>
    <mn>2</mn>
    <mrow>
      <mi>p</mi>
      <mi>e</mi>
      <mi>a</mi>
      <mi>r</mi>
      <mi>s</mi>
    </mrow>
    <mo>=</mo>
    <mn>20</mn>
    <mrow>
      <mrow>
        <mi mathvariant="normal">k</mi>
      </mrow>
      <mrow>
        <mi mathvariant="normal">r</mi>
      </mrow>
    </mrow>
  </mrow>
</math>
```

**Comment by ronkok** [comment]
I think it is poor practice to write a multi-character variable name in italic. Inside a math expression, italic letters are single-letter variable names. Because of that convention, you can read an equation like $E = mc^2$ and know that the $m$ and the $c$ are two variables, not one. Yes, I know many people on the web write multi-character variable names in italic. The world would be a more legible place if they stopped doing that.

Which is why I am not going to consolidate multiple letters inside `\mathit` into a single `<mi>` element.

You also bring up `\mathrm` and there you have a good point. I agree with [this opinion](https://tex.stackexchange.com/a/482988) that `\mathrm` is a good way to write multi-character variable names. I'll do as you request and consolidate `\mathrm` terms into a single `<mi>` element. Note that there will be exceptions. If people get wacky and write fractions inside `\mathrm`, Temml will give up and render each character separately.

Note that Temml already consolidates `\text{…}` into a single element. But that is a `<mtext>` element, not a `<mi>`. If your example uses `kr` to convey a currency name, then maybe it should be `\text{kr}.`

A week or two may go by before I get to this.

**Comment by Carlosted** [comment]
I'm just a stranger who stubbled on this issue randomly, but doesn't `\operatorname{kr}` follow the expected behavior of `\mathrm{kr}`?

I agree with @ronkok on the fact that variables should always be 1 character, and that is the way the TeXBook _and_ the `mathml` spec expect things to be. A `<mi>` with more than one character usually represent a function name such as `sin`, `exp`, etc., and typesets them with the roman font by default.

On a sidenote, the `mathml-core` `<ms>` tag could be used as well for the same purpose.

**Comment by Leon-Lj** [comment]
> I agree with @ronkok on the fact that variables should always be 1 character, and that is the way the TeXBook _and_ the `mathml` spec expect things to be. 

On a university level, yes. On a lower to mid school level? You start out with `🍏+🍐`, move on to `apple+pear`, and then eventually you learn that you've been doing `x+y` all along.

I agree that multi char variables should be in roman, and as @Carlosted points out, that's the default behaviour for multi char `<mi>`-tags ([MDN](https://developer.mozilla.org/en-US/docs/Web/MathML/Element/mi)). But I do think that groupings should remain when translated, so `\mathit{apple}` and` \mathrm{apple}` could both translate to `<mi>apple</mi>`. Then it'll render correctly as roman `apple`.

However, I was a bit off the mark in the original post. NVDA + MathCAT (mathml plugin) does actually read the `\mathit` part as one word, because it's in one `<mrow>`. It tries to be smart about it, if it looks like a word it'll read it as a word, if it doesn't it'll spell it out. 

So for example this will read as `apple`.
```
<mrow>
  <mi>a</mi>
  <mi>p</mi>
  <mi>p</mi>
  <mi>l</mi>
  <mi>e</mi>
</mrow>
```

While this reads as `v w x y z`
```
<mrow>
  <mi>v</mi>
  <mi>w</mi>
  <mi>x</mi>
  <mi>y</mi>
  <mi>z</mi>
</mrow>
```

It can be fooled, but overall it does a pretty good job of knowing when to spell it out and when to read it as a word ([some examples](https://codepen.io/leon-lj/pen/poBJrMd)). Of course it could differ from screen reader to screen reader, how well they handle knowing when to spell it out and when to read it as a word.

So `\mathit` works pretty ok. The problem with the mathml that temml produces is with the `\mathrm`, since it splits each letter into its own `<mrow>`. So for example the mathml from `\mathit{apple}+\mathrm{pear}` will be read by NVDA + MathCAT as `apple + p e a r`
```
<math display="block" class="tml-display" style="display:block math;">
  <mrow>
    <mrow>
      <mi>a</mi>
      <mi>p</mi>
      <mi>p</mi>
      <mi>l</mi>
      <mi>e</mi>
    </mrow>
    <mo>+</mo>
    <mrow>
      <mrow>
        <mi mathvariant="normal">p</mi>
      </mrow>
      <mrow>
        <mi mathvariant="normal">e</mi>
      </mrow>
      <mrow>
        <mi mathvariant="normal">a</mi>
      </mrow>
      <mrow>
        <mi mathvariant="normal">r</mi>
      </mrow>
    </mrow>
  </mrow>
</math>
```

**Comment by ronkok** [comment]
@Carlosted, Temml's version of `\operatorname` differs from `\mathrm` in two respects:

1. `\operatorname` spacing is similar to math operators such as `\sin`. There is a small space placed between the operator and a subsequent variable, as in $\sin x$.
2. Temml's version of `\operatorname` writes an invisible `<mo>&ApplyFunction;</mo>` after the `\operatorname{…}`. This is done to enhance  accessibility. In MathML, variable names and function names are both written in an `<mi>` element. The `<mo>&ApplyFunction;</mo>` enables a screen reader to recognize a function name, so it can say "sin of" instead of "sin times".

As noted above, Temml's `\mathrm` does not yet consolidate its contents into a single `<mi>` element. That will change next week.

**Comment by ronkok** [comment]
Resolved by release v0.10.23.

**Comment by davidcarlisle** [comment]
> Which is why I am not going to consolidate multiple letters inside \mathit into a single <mi> element.

I'm sorry but I think from both a LaTeX and MathML perspective that is an unfortunate choice. In LaTeX `$\mathit{Diff}$` is explicitly different from `$Diff$` 

 the former uses a text italic font including inter-letter kerns and ff ligatures to make the single word identifier _Diff_  

the latter uses the math italic font which has wide sidebearings and no ff ligatures to represent the product _D i f f_

While classical mathematical variables are typically single letter, category theory often needs differently styled multi-letter identifiers, as does mathematical notation used for pseudo-code in computer science. 

So the only reason to use `\mathit` in a TeX source is to denote a multi-letter identifier.

```
\documentclass{article}

\begin{document}

$\mathit{Diff}+Diff$

\end{document}
```

<img width="384" height="146" alt="Image" src="https://github.com/user-attachments/assets/0f372013-e8bf-426d-a5fc-64bbb88309d0" />

**Comment by ronkok** [comment]
Okay, you've persuaded me. I'll consolidate `\mathit` into a single `<mi>` element. The next release will be sometime later this week.

**Comment by davidcarlisle** [comment]
thanks

**Comment by ronkok** [comment]
Resolved by release v0.13.3.
